### DISCUSSION TITLE
Re-process Google Takeout folders | Remove partner images etc.

### DISCUSSION THREAD

**Discussion by therealmelee** [simulot/immich-go#85] [Ideas]
Hey folks.

Love this project!

I imported / uploaded my Takeout folder whole, and later have come across the Google Photos options, including the one to import Partner photos.

Is there a way to re-process the folder using a later version of immich-go and have either the tool itself or the API know to remove/mark files that I now would NOT want to have included?

Alternatively, does immich-go import the Partner flag into the Immich database itself (so I could filter within immich and remove there), or only use it for the purposes of the flag to skip?

Rescanning would also help for instance when a release like 0.8.6 comes along that improves the number of images imported, to be able to go through again and ensure everything was properly attached.

Thanks in advance for your help!

**Comment by simulot** [comment]
Thanks for using the project

Also immich-go is meant to import photos... not remove them. I haven't even envisage it. 
But why not. I have started the building of a 'tool' command for this kind of process.
Let's move this discussion to 'ideas' categories. 

As far as I known, Immich doesn't have flag to tag photos as partner's ones.

**Reply by therealmelee** [reply]
Totally understand and certainly don't mean to cause distraction or too much scope creep. And I'll say up front I'm not an Engineer and don't have any more than a hobbiest understanding of this, but I can try to use plain English 👍 

I think the wider use-case is in re-processing a Takeout library as the feature set broadens; and/or for exploiting flags that immich-go detects that Immich itself doesn't.

In this case - looks like you're setting or detecting a FromPartner boolean from the input files somehow (inside a json?), and that could be useful at a later date. For example, I've imported my full Takeout library, but now this feature exists.

I could (1) delete my entire library and start over - including processing time to regenerate all thumbnails etc..; or
(2) Do something like this badly written SQL, which could be achieved through manipulation of the database itself - not sure if it's possible to do via the API (maybe could leverage the existing 'already exists' matching to figure that out?)

SELECT *
FROM [immich library]

LEFT JOIN
(
SELECT *
FROM [takeout library]
WHERE FromPartner = TRUE
)

ON a.ID = b.ID (hash or ID).

Now we've matched based on that boolean, and can delete the matches from the Immich library. Or perhaps we want to add the 'Partner Library' at a later time - we could do the same, instead of deleting, it creates an Album with those images.

**Reply by therealmelee** [reply]
Super mega duper bonus feature idea, would be to give immich-go multiple API keys, and have it deduplicate the libraries between them - sharing pictures between the two+ users but only storing one copy.

Until then I think fclones is the efficient solution - it can quickly and accurately detect matches and replace them with hardlinks to free up space. That is - unless the images themselves are modified upon import into Immich and so wouldn't match each other anymore?

**Reply by simulot** [reply]
Great idea! dedup accros users!  

However this should be implemented in immich directly
